pkg/objstore/s3/s3.go

```diff
@@ -452,7 +452,7 @@ func (b *Bucket) Upload(ctx context.Context, name string, r io.Reader) error {
 	// partSize cannot be larger than object size.
 	partSize := b.partSize
 	if size < int64(partSize) {
-		partSize = 0
+		partSize = size // set partSize to the object size
```

Comment:
I think we wanted something else here. PartSize = 0 was meaning we don't want multipart.